applications/FemToDemApplication/custom_processes/compute_normalized_free_energy_on_nodes_process.cpp

```diff
@@ -0,0 +1,335 @@
+        double& r_norm = it_node->GetSolutionStepValue(EQUIVALENT_NODAL_STRESS);
+        r_norm = nodal_free_energy;
+
+        if (mCorrectWithDisplacements) {
+            const double r_displ = norm_2(it_node->GetSolutionStepValue(DISPLACEMENT));
+            r_norm *= r_displ;
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+double ComputeNormalizedFreeEnergyOnNodesProcess::CalculateNormalizedFreeEnergy(
+    const Vector& rStrainVector, 
+    const Vector& rStressVector, 
+    const double Damage, 
+    const Properties& rMatProps,
+    Geometry<Node<3>>& rGeometry
+    )
+{
+    double normalized_free_energy;
+
+    normalized_free_energy = inner_prod(rStrainVector, rStressVector);
+    if (mComputeNormalizedFreeEnergy) {
+        const double fracture_energy_tension = rMatProps[FRAC_ENERGY_T];
+        const double yield_tension = rMatProps[YIELD_STRESS_T];
+        const double yield_compression = rMatProps[YIELD_STRESS_C];
+        const double ratio = yield_compression / yield_tension;
+        const double fracture_energy_compression = fracture_energy_tension * std::pow(ratio, 2.0);
+        const double density = rMatProps[DENSITY];
+        
+        KRATOS_ERROR_IF(yield_tension < tolerance) << "The YIELD_TENSION is null or not assigned" << std::endl;
+        KRATOS_ERROR_IF(yield_compression < tolerance) << "The YIELD_COMPRESSION is null or not assigned" << std::endl;
+        KRATOS_ERROR_IF(fracture_energy_tension < tolerance) << "The FRACTURE ENERGY is null or not assigned" << std::endl;
+        KRATOS_ERROR_IF(density < tolerance) << "The DENSITY is null or not assigned" << std::endl;
+
+        double characteristic_length, r;
+        if (mDimension == 2) {
+            characteristic_length = this->CalculateCharacteristicLength2D(rGeometry);
+            r = this->ComputeTensionFactor2D(rStressVector);
+        } else {
+            characteristic_length = this->CalculateCharacteristicLength3D(rGeometry);
+            r = this->ComputeTensionFactor3D(rStressVector);
+        }
+        const double g_t = fracture_energy_tension / characteristic_length;
+        const double g_c = fracture_energy_compression / characteristic_length;
+        normalized_free_energy *= (1.0 - Damage);
+        normalized_free_energy /= (2.0 * density);
+        normalized_free_energy *= (r / g_t + (1.0 - r) / g_c);
+    }
+    return normalized_free_energy;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+double ComputeNormalizedFreeEnergyOnNodesProcess::CalculateCharacteristicLength2D(
+    const Geometry<Node<3>>& rGeometry
+    )
+{
+    Vector node_1_coordinates = rGeometry[0].Coordinates();
```

Comment:
Coordinates are an array, you are doing an static cast here, and a copy
![imagen](https://user-images.githubusercontent.com/19991680/35767644-8a9c6856-08f0-11e8-9944-e72c6c1db7b3.png)